spartan/aztec-network/templates/validator.yaml

```diff
@@ -19,39 +19,73 @@ spec:
         app: validator
     spec:
       initContainers:
-        - name: wait-for-boot-node
+        - name: wait-for-services
           image: {{ .Values.images.curl.image }}
           command:
             - /bin/sh
             - -c
             - |
+              # First check ethereum node
               until curl -s -X POST -H 'Content-Type: application/json' \
                 -d '{"jsonrpc":"2.0","method":"web3_clientVersion","params":[],"id":67}' \
                 {{ include "aztec-network.ethereumHost" . }} | grep -q reth; do
                 echo "Waiting for Ethereum node..."
                 sleep 5
               done
               echo "Ethereum node is ready!"
+
               {{- if .Values.telemetry.enabled }}
               until curl --head --silent {{ include "aztec-network.otelCollectorMetricsEndpoint" . }} > /dev/null; do
                 echo "Waiting for OpenTelemetry collector..."
                 sleep 5
               done
               echo "OpenTelemetry collector is ready!"
               {{- end }}
-              until curl --head --silent {{ include "aztec-network.bootNodeUrl" . }}/status; do
-                echo "Waiting for boot node..."
-                sleep 5
-              done
-              echo "Boot node is ready!"
+
+              if [ "{{ .Values.validator.dynamicBootNode }}" = "true" ]; then
+                # Get the list of pod IPs for the validator service
+                MAX_ATTEMPTS=3
+                for i in $(seq 0 $(({{ .Values.validator.replicas }} - 1))); do
+                  PEER_IP="{{ include "aztec-network.fullname" . }}-validator-${i}.{{ include "aztec-network.fullname" . }}-validator"
+                  echo "Checking ${PEER_IP} for /status"
+                  for attempt in $(seq 1 $MAX_ATTEMPTS); do
+                    if curl --silent --head --fail "http://${PEER_IP}:{{ .Values.validator.service.nodePort }}/status" > /dev/null; then
+                      echo "Found responsive peer at ${PEER_IP}"
+                      # the PXE has its node set to the the validator service.
+                      # and that's all we need to know to bootstrap,
+                      # since it will get a good node ENR from whatever node the PXE connects to.
+                      echo "{{ include "aztec-network.pxeUrl" . }}" > /shared/pxe_url
```

Comment:
As noted in the comment above, this looping is really just a safety/sanity check, since as long as one validator is up, the PXE *should* be routed to it.